src/test/kotlin/service/user/RegisterTest.kt

```diff
@@ -0,0 +1,46 @@
+package se.uulm.snowballr.backend.service.user
+
+import io.mockk.coEvery
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import se.uulm.snowballr.backend.DataBuilder
+import se.uulm.snowballr.backend.service.MainServiceTest
+import se.uulm.snowballr.backend.testCoroutine
+import snowballr.Authentication
+
+@ExperimentalCoroutinesApi
+@DelicateCoroutinesApi
+internal class RegisterTest : MainServiceTest() {
+    @Test
+    fun `When a user is registered, then no exception is thrown`() = testCoroutine {
+        val request = Authentication.RegisterRequest.newBuilder().build()
+        val user = DataBuilder.createExampleUser()
+
+        coEvery { userRepoMock.doesUserExistByEmail(any()) } returns false
+        coEvery { userRepoMock.createUser(any(), any()) } returns user
+
+        assertDoesNotThrow { mainService.register(request) }
+    }
+
+    @Test
+    fun `When a user already exists, then an exception is thrown`() = testCoroutine {
+        val request = Authentication.RegisterRequest.newBuilder().build()
+
+        coEvery { userRepoMock.doesUserExistByEmail(any()) } returns true
+
+        assertThrows<Exception> { mainService.register(request) }
```

Comment:
Check for the specific exception. Otherwise, this could be any exception, and we wouldn't know if this test actually tests the case we want to test.
```suggestion
        assertThrows<SnowballRException.DuplicateEntityException.UserEmail> { mainService.register(request) }
```

Reply:
Makes sense.